pypmc/sampler/markov_chain.py

```diff
@@ -132,27 +137,31 @@ def run(self, N=1):
             log_rho = get_log_rho(proposed_point, proposed_eval)
 
             # check for NaN
-            if _np.isnan(log_rho): raise ValueError('encountered NaN')
-
-            # accept if rho = 1
-            if log_rho >=0:
-                accept_count += 1
-                this_run[i_N]            = proposed_point
-                self.current_point       = proposed_point
-                self.current_target_eval = proposed_eval
-
-            # accept with probability rho
-            elif log_rho >= _np.log(self.rng.rand()):
-                accept_count += 1
-                this_run[i_N]            = proposed_point
-                self.current_point       = proposed_point
-                self.current_target_eval = proposed_eval
-
-            # reject if not accepted
-            else:
+            if _np.isnan(log_rho):
+                # raise error if so desired
+                if not continue_on_NaN: raise ValueError('encountered NaN')
+                # otherwise reject
```

Comment:
From the small excerpt I see on github, it looks like `log_rho` is not used anywhere else and the proposed point is rejected, so no harm done. But it would be a bit cleaner IMHO to explictly go to the reject part. Would be a candidate for a `goto` :) But no such keyword in python.

@dvandyk did you run your code locally with this change? I would like to take a 2nd look if no harm is done by `log_rho= NaN` somewhere down the line

Reply:
@fredRos Yes, I did run the code locally and it works for my purpose. It also passes the tests.
```log_rho``` is used later in deciding whether to reject regularly.